homeassistant/components/vesync/light.py

```diff
@@ -68,17 +88,105 @@ def turn_on(self, **kwargs):
             brightness = round((brightness / 255) * 100)
             # clamp to 1-100
             brightness = max(1, min(brightness, 100))
-            self.dimmer.set_brightness(brightness)
+            self.device.set_brightness(brightness)
         # Avoid turning device back on if this is just a brightness adjustment
         if not self.is_on:
             self.device.turn_on()
 
     @property
-    def supported_features(self):
-        """Get supported features for this entity."""
-        return SUPPORT_BRIGHTNESS
+    def color_mode(self):
+        """Set color mode for this entity."""
+        return COLOR_MODE_BRIGHTNESS
 
     @property
-    def brightness(self):
-        """Get dimmer brightness."""
-        return round((int(self.dimmer.brightness) / 100) * 255)
+    def supported_color_modes(self):
+        """Flag supported color_modes (in an array format)."""
+        return [COLOR_MODE_BRIGHTNESS]
+
+
+class VeSyncTunableWhiteLightHA(VeSyncBaseLight, LightEntity):
+    """Representation of a VeSync Tunable White Light device."""
+
+    def __init__(self, device):
+        """Initialize the VeSync Tunable White Light device."""
+        super().__init__(device)
+        self.device = device
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            # get brightness from HA data
+            brightness = int(kwargs[ATTR_BRIGHTNESS])
+            # convert to percent that vesync api expects
+            brightness = round((brightness / 255) * 100)
+            # clamp to 1-100
+            brightness = max(1, min(brightness, 100))
+            # pass value to pyvesync library api
+            self.device.set_brightness(brightness)
+
```

Comment:
```suggestion
        if ATTR_BRIGHTNESS in kwargs:
            # get brightness from HA data
            brightness = int(kwargs[ATTR_BRIGHTNESS])
            # convert to percent that vesync api expects
            brightness = round((brightness / 255) * 100)
            # clamp to 1-100
            brightness = max(1, min(brightness, 100))
            # pass value to pyvesync library api
            self.device.set_brightness(brightness)

```

Can you avoid the duplicate brightness code by calling `super().turn_on(**kwargs)` after checking `ATTR_COLOR_TEMP` and dropping this block?

Reply:
I started doing just what you said with super(), then I realised since the color_temp logic is safe contained inside a IF condition, I moved the whole block to the base class. making the whole file cleaner with no repeated code at all.